typify-impl/src/lib.rs

```diff
@@ -237,38 +237,80 @@ pub(crate) enum DefaultImpl {
     NZU64,
 }
 
+/// Type name to use in generated code.
+#[derive(Clone)]
+pub struct MapType(pub syn::Type);
```

Comment:
What's the benefit of a newtype here over using syn::Type?

Reply:
A newtype allowed me to implement `Default` correctly. :)

Reply:
Also `serde` implementations to enable parsing.